### ISSUE TITLE
Entire row right click

### ISSUE THREAD

**Issue by ecraven** [ocornut/imgui#769]
I'm creating a column like this:
``
ImGui::Columns(2, "mycolumns", false);
for(....) {
ImGui::BeginGroup();
ImGui::Selectable(...); ImGui::NextColumn();
ImGui::Text(...); ImGui::NextColumn();
ImGui::EndGroup();
if(ImGui::IsItemClicked(1)) {
// right click onto row
}
}
``

However, `ImGui::IsItemClicked` only registers when clicking onto the first column, not the second.. am I using groups incorrectly here? How do I best achieve this?

**Comment by ocornut** [comment]
The intention is correct here, but `IsItemClicked`/`IsItemHovered` takes account of the current clipping rectangle, which at the time of moving back to the first column will only cover those. 

In this specific situation the simplest is just to call Selectable with the `ImGuiSelectableFlags_SpanAllColumns` flag, maybe that'll just do what you need here?

**Comment by ecraven** [comment]
That flag is actually part of the code (I just neglected to actually put it into the pseudocode).
I think I'm looking for a general way to create "compound" widgets, as detailed in #775. This is just a special case of that. Something like BeginSelectable() / EndSelectable() for example :-/

**Comment by graysmiley** [comment]
Is it desirable to add an `ImGuiSelectableFlags_AllowRightClick` flag to solve this? I've implemented this for my own use by adding the following to imgui_widgets.cpp:
`if (flags & ImGuiSelectableFlags_AllowRightClick)   { button_flags |= ImGuiButtonFlags_PressedOnClickRelease | ImGuiButtonFlags_MouseButtonRight | ImGuiButtonFlags_MouseButtonLeft;`

One can then check if a row was right clicked with the following:
`if (ImGui::Selectable(<label>, <selected>, ImGuiSelectableFlags_SpanAllColumns | ImGuiSelectableFlags_AllowRightClick))
{
    if (ImGui::IsMouseReleased(ImGuiMouseButton_Right))
    {
        // Rick click pressed on current item
    }
}`

**Comment by ocornut** [comment]
You shouldn't need it, just telling for `IsItemHovered() && IsMouseReleased(ImGuiMouseButton_Right)` should work.

**Comment by graysmiley** [comment]
You're correct. Adding that check after Selectable(...) will produce the desired behavior. Thanks for the info! I've closed the pull request.
